core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala

```diff
@@ -30,6 +30,8 @@ private[spark] class ApplicationDescription(
 
   val user = System.getProperty("user.name", "<unknown>")
 
+  var coreNumPerTask = 1
```

Comment:
This field is somewhat arbitrary. A better place to put it would be in the constructor. Also I would rename it to be `coresPerTask`